### PR TITLE
Add api to show hide series without removing it #471

### LINKED ISSUE #471: Add API to show/hide series without removing it
I mean to show a line series legend somewhere at the left top corner of a plot and to have an abillity to turning visibility on and off to some of them. What about such feature?

### PR DESCRIPTION
**Pull request by olegchernenko**
**Type of PR:** 

**PR checklist:**

- [x] Addresses an existing issue: fixes #471
- [x] Includes tests
- [x] Documentation update

**Overview of change:**

**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

File: src/views/pane/series-markers-pane-view.ts [hunk 1]
```diff
@@ -167,6 +168,7 @@ export class SeriesMarkersPaneView implements IUpdatablePaneView {
 		const layoutOptions = this._model.options().layout;
 
 		this._data.visibleRange = null;
+		this._data.visible = this._series.options().visible;
```
Comment (timocov) [inline src/views/pane/series-markers-pane-view.ts:171]: why do we need to save it in data?

File: src/views/pane/line-pane-view.ts [hunk 2]
```diff
@@ -27,6 +27,7 @@ export class SeriesLinePaneView extends LinePaneViewBase<'Line', LineItem> {
 			lineType: lineStyleProps.lineType,
 			lineWidth: lineStyleProps.lineWidth,
 			visibleRange: this._itemsVisibleRange,
+			visible: lineStyleProps.visible,
```
Comment (timocov) [inline src/views/pane/line-pane-view.ts:30]: why do we need to add it to renderer options? I guess it's the last guy who should know about that

File: src/views/pane/histogram-pane-view.ts [hunk 3]
```diff
@@ -96,6 +97,7 @@ export class SeriesHistogramPaneView extends SeriesPaneViewBase<'Histogram', Tim
 		this._histogramData.histogramBase = histogramBase;
 		this._histogramData.visibleRange = visibleTimedValues(this._histogramData.items, visibleBars, false);
 		this._histogramData.barSpacing = barSpacing;
+		this._histogramData.visible = this._series.options().visible;
```
Comment (timocov) [inline src/views/pane/histogram-pane-view.ts:100]: the same

File: docs/series-basics.md [hunk 4]
```diff
@@ -141,6 +142,27 @@ chart.applyOptions({
 });
 ```
 
+### Visibility
+
+When adding any series to a chart, you can specify its visibility using `visible` property.
+
+By default, created series is visible. You can change its visibility using `SeriesApi.applyOptions`:
```
Comment (timocov) [inline docs/series-basics.md:149]:
```suggestion
By default, all created series are visible. You can change the visibility using `applyOptions` method:
```

File: docs/series-basics.md [hunk 5]
```diff
@@ -141,6 +142,27 @@ chart.applyOptions({
...
+
+```js
+series.applyOptions({
+    visible: false,
+});
+```
+
+When you hide a series - price line, base line, price labels and markers of series will also be hidden.
```
Comment (timocov) [inline docs/series-basics.md:157]:
```suggestion
If the series is hidden - everything including its price lines, base line, price labels and markers will be hidden as well.
```

File: docs/series-basics.md [hunk 6]
```diff
@@ -141,6 +142,27 @@ chart.applyOptions({
...
+Note that hiding a series is not equivalent to deleting it, you can show it again:
+
+```js
+series.applyOptions({
+    visible: true,
+});
+```
```
Comment (timocov) [inline docs/series-basics.md:164]:
```suggestion
Note that hiding the series is not equivalent to deleting it - hiding does not affect the time scale at all in opposite of deleting, when the time scale might be changed (some points might be deleted).
```

File: docs/series-basics.md [hunk 7]
```diff
@@ -233,7 +255,7 @@ You can set the width, style and color of this line or disable it using the foll
 
 |Name|Type|Default|Description|
 |----|----|-------|-|
-|`priceLineVisible`|`boolean`|`true`|If true, a series' price line is displayed on a chart|
+|`priceLineVisible`|`boolean`|`true`|If true, a series' price line is displayed on a chart if a series is visible|
```
Comment (ezhukovskiy) [inline docs/series-basics.md:258]:
```suggestion
|`priceLineVisible`|`boolean`|`true`|If true, a series' price line is displayed on a chart if the series is visible|
```

File: docs/series-basics.md [hunk 8]
```diff
@@ -257,7 +279,7 @@ There is an option to hide it as well.
 
 |Name|Type|Default|Description|
 |----|----|-------|-|
-|`lastValueVisible`|`boolean`|`true`|If true, the label with the current price is displayed on the price scale|
+|`lastValueVisible`|`boolean`|`true`|If true, the label with the current price is displayed on the price scale if a series is visible|
```
Comment (ezhukovskiy) [inline docs/series-basics.md:282]:
```suggestion
|`lastValueVisible`|`boolean`|`true`|If true, the label with the current price is displayed on the price scale if the series is visible|
```

File: docs/series-basics.md [hunk 9]
```diff
@@ -274,7 +296,7 @@ You can set the width, style and color of this line or disable it using the foll
 
 |Name|Type|Default|Description|
 |----|----|-------|-|
-|`baseLineVisible`|`boolean`|`true`|If true, a series' base line is displayed on a chart|
+|`baseLineVisible`|`boolean`|`true`|If true, a series' base line is displayed on a chart if a series is visible|
```
Comment (ezhukovskiy) [inline docs/series-basics.md:299]:
```suggestion
|`baseLineVisible`|`boolean`|`true`|If true, a series' base line is displayed on a chart if the series is visible|
```

**Review by ezhukovskiy — CHANGES_REQUESTED** [review]

**Review by timocov — APPROVED** [review]

**Comment by timocov** [comment]
It seems that magnet doesn't respect `visible` option (perhaps something else too?).

**Review by timocov — APPROVED** [review]

**Pull request merged**